staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -273,13 +273,13 @@ func (m *Helper) Patch(namespace, name string, pt types.PatchType, data []byte,
 func (m *Helper) Replace(namespace, name string, overwrite bool, obj runtime.Object) (runtime.Object, error) {
 	c := m.RESTClient
 	var options = &metav1.UpdateOptions{}
-	if m.ServerDryRun {
+	if m.ServerDryRun && len(options.DryRun) == 0 {
 		options.DryRun = []string{metav1.DryRunAll}
 	}
-	if m.FieldManager != "" {
+	if m.FieldManager != "" && options.FieldManager == "" {
 		options.FieldManager = m.FieldManager
 	}
-	if m.FieldValidation != "" {
+	if m.FieldValidation != "" && options.FieldValidation == "" {
```

Comment:
This is a lot of copy-paste, all in the same helper. Does the helper need some help?

Reply:
That sure would be something but unfortunately the types are all different: `metav1.DeleteOptions`, `metav1.PatchOptions`, `metav1.UpdateOptions`. I'm not sure how to share the code for this in Go

Reply:
sad face emoji